sentry-android-fragment/src/main/java/io/sentry/android/fragment/SentryFragmentLifecycleCallbacks.kt

```diff
@@ -0,0 +1,91 @@
+package io.sentry.android.fragment
+
+import android.content.Context
+import android.os.Bundle
+import android.view.View
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentManager
+import androidx.fragment.app.FragmentManager.FragmentLifecycleCallbacks
+import io.sentry.Breadcrumb
+import io.sentry.IHub
+import io.sentry.SentryLevel.INFO
+
+@Suppress("TooManyFunctions")
+internal class SentryFragmentLifecycleCallbacks(private val hub: IHub) : FragmentLifecycleCallbacks() {
+
+    override fun onFragmentAttached(
+        fragmentManager: FragmentManager,
+        fragment: Fragment,
+        context: Context
+    ) {
+        addBreadcrumb(fragment, "attached")
+    }
+
+    override fun onFragmentSaveInstanceState(
+        fragmentManager: FragmentManager,
+        fragment: Fragment,
+        outState: Bundle
+    ) {
+        addBreadcrumb(fragment, "save instance state")
+    }
+
+    override fun onFragmentCreated(
+        fragmentManager: FragmentManager,
+        fragment: Fragment,
+        savedInstanceState: Bundle?
+    ) {
+        addBreadcrumb(fragment, "created")
+    }
+
+    override fun onFragmentViewCreated(
+        fragmentManager: FragmentManager,
+        fragment: Fragment,
+        view: View,
+        savedInstanceState: Bundle?
+    ) {
+        addBreadcrumb(fragment, "view created")
+    }
+
+    override fun onFragmentStarted(fragmentManager: FragmentManager, fragment: Fragment) {
+        addBreadcrumb(fragment, "started")
+    }
+
+    override fun onFragmentResumed(fragmentManager: FragmentManager, fragment: Fragment) {
+        addBreadcrumb(fragment, "resumed")
+    }
+
+    override fun onFragmentPaused(fragmentManager: FragmentManager, fragment: Fragment) {
+        addBreadcrumb(fragment, "paused")
+    }
+
+    override fun onFragmentStopped(fragmentManager: FragmentManager, fragment: Fragment) {
+        addBreadcrumb(fragment, "stopped")
+    }
+
+    override fun onFragmentViewDestroyed(fragmentManager: FragmentManager, fragment: Fragment) {
+        addBreadcrumb(fragment, "view destroyed")
+    }
+
+    override fun onFragmentDestroyed(fragmentManager: FragmentManager, fragment: Fragment) {
+        addBreadcrumb(fragment, "destroyed")
+    }
+
+    override fun onFragmentDetached(fragmentManager: FragmentManager, fragment: Fragment) {
+        addBreadcrumb(fragment, "detached")
+    }
+
+    private fun addBreadcrumb(fragment: Fragment, state: String) {
+        val breadcrumb = Breadcrumb().apply {
+            type = "navigation"
+            setData("state", state)
+            setData("screen", getFragmentName(fragment))
+            category = "ui.lifecycle"
```

Comment:
should we be a bit more specific here to differentiate between Activity and fragment? like `fragment.lifecycle` or `ui.fragment.lifecycle`?

Reply:
Sure, good idea! I like ui.fragment.lifecycle for the category name

Reply:
Done in b673c00d95f97381da577259a043d6e6eaf7b142